duckduckgo/Android
# Fire button education

Task/Issue URL: https://app.asana.com/0/488551667048375/1190516087603930/f
Tech Design URL: 
CC: https://app.asana.com/0/72649045549333/1186511372568850/f

**Description**:
We would like to encourage users to try the fire button during onboarding.

This PR adds an experiment to test a new onboarding flow where we will ask the user to try Fire button.

For new users, after they see the first in-browser CTA (except SERP CTA), we will start highlighting the Fire button with a pulse animation. That animation will last till the user interacts with the button.
The first time the user interacts with the Fire button, we will show a CTA to encourage them to try clearing all data.
Fire button education variant also includes a Fire animation.

**Steps to test this PR**:

---- Fire education variant ----

>Test 1 -Full flow-: 
1. Hardcode `zr` variant
1. Perform fresh install
1. Navigate to Browser screen
1. Visit SERP Page
1. Ensure after dismissing SERP CTA, Fire button is not highlighted
1. Visit any other site
1. Ensure after dismissing CTA, Fire button is highlighted
1. Click on the Fire button
1. Ensure Fire dialog displayed and CTA appears: `Now tap below to burn your browsing history...`
1. Click on the "Clear All Tabs and Data"
1. Ensure Fire animation runs and app restarts

>Test 2 -Try Fire button before animation started-: 
1. Hardcode `zr` variant
1. Perform fresh install
1. Navigate to Browser screen
1. Click on the Fire button
1. Ensure Fire dialog displayed and CTA appears: `Now tap below to burn your browsing history...`
1. Dismiss dialog (go back to the browser)
1. Visit any non-SERP site
1. Ensure after dismissing CTA, Fire button is **not** highlighted

>Test 3 -Pulse animation always active till user interacts with it-: 
1. Hardcode `zr` variant
1. Perform fresh install
1. Navigate to Browser screen
1. Visit any non-SERP site
1. Ensure after dismissing CTA, Fire button is highlighted
1. Kill the app / send the app to background
1. Go back to the app
1. Ensure Fire button is still highlighted

>Test 4 -Hide all tips before fire education-: 
1. Hardcode `zr` variant
1. Perform fresh install
1. Navigate to Browser screen
1. Visit SERP Page
1. When CTA appears, **click on Hide all tips**
1. Visit any other site
1. Ensure no CTA and the Fire button is **not** highlighted
1. Click on the Fire button
1. Ensure Fire dialog displayed but no CTA shown
1. Click on the "Clear All Tabs and Data"
1. Ensure Fire animation runs and app restarts

>Test 5 -Edge case: Avoid browser CTA appearing on top of Fire dialog-: 
1. Hardcode `zr` variant
1. Perform fresh install
1. Navigate to Browser screen
1. Visit any site and **quickly** click on the Fire button
1. Wait till website finishes loading in the background
1. Ensure no browser CTA has appeared on top of the Fire dialog

---- Control variant: No Fire button education ----
>Test A -Full flow-: 
1. Hardcode `zm` variant
1. Perform fresh install
1. Navigate to Browser screen
1. Visit SERP Page
1. Ensure after dismissing SERP CTA, Fire button is **not** highlighted
1. Visit any other site
1. Ensure after dismissing CTA, Fire button is **not** highlighted
1. Click on the Fire button
1. Ensure Fire dialog displayed without any CTA
1. Click on the "Clear All Tabs and Data"
1. Ensure Fire animation did not run

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
